driver/compile_common.ml

```diff
@@ -43,6 +43,7 @@ let init ppf_dump ~native ~tool_name ~source_file ~output_prefix =
     source_file;
     ppf_dump;
     tool_name;
+    native;
```

Comment:
With the final state of the PR, `Compile_common.interface` takes an `info` parameter instead of cooking it itself, so this problem does not occur (Compile calls with native=false, Optcompile with native=true), so this problem doesn't exist anymore.